L1T phase-2: Add the Phase2 L1T VertexFinder

cecilecaillol opened the pull request:
#### PR description:
This pull request includes:

An EDProducer that implements the primary-vertex-finding algorithms investigated for the L1T Interim TDR.
The EDAnalyser that was used to study the performance of these algorithms.
The Ntupler for analysis of vertexing algorithms.

The code for all of these is fully contained within L1Trigger/VertexFinder, and the l1tVertexFinder namespace; however of course in principle the name of this directory & namespace could be changed to conform to standard conventions.

A few more details on the EDProducer ...

Source code: interface/VertexProducer.h & plugins/VertexProducer.cc
It creates a l1t::VertexCollection, which is based on the L1T vertex data format class
l1vertices - vector of all vertices found by the algo defined in the Python config (default config uses "DBSCAN" algo)
Python config fragment (to be used by default for downstream algos): python/VertexProducer_cff.py
Parameter values are set to run DBSCAN vertex-finding algo
Other vertexing-finding algorithms can be run by changing the value of the Algorithm parameter in the VertexReconstruction parameter set.

Apart from that EDProducer, the other files in this pull request are:

Header + .cc files for AlgoSettings, L1Track, RecoVertex and `VertexFinder classes that are used by the vertex producer itself
interface/selection.h & src/selection.cc : Contains implementation of getPrimaryVertex function, that currently identifies highest-sum-pT vertex in vector of vertex objects.
plugins/VertexAnalyzer.cc : Implements the EDAnalyzer that we use to study the performance of vertex-finding algorithms
Header + .cc files for classes used by the EDAnalyzer: L1TrackTruthMatched, RecoVertexWithTP, Stub & TP


#### PR validation:

This code had undergone clang-tidy and clang-format checks. It has been validated to make sure it performs as it did on the CMSSW_10_1_5 integration branch. There are still some performance issues (speed) with the EDAnalyzer and ntuple maker, but these can be worked out in a later PR. The core functionality of the vertexing (EDProducer) performs quite well.

Rebasing of https://github.com/cms-l1t-offline/cmssw/pull/874

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Rebasing of https://github.com/cms-l1t-offline/cmssw/pull/874

emacdonald16 committed:
Track-objects in CMSSW11; included prompt vs displaced track flags

(cherry picked from commit dad89184df2d1a46a79512186afa8bf27f27183e)

emacdonald16 committed:
name changes for CMS guidelines

(cherry picked from commit 3eb74bb80c19fd2d510f5e41248f6f16256c4103)

emacdonald16 committed:
minor name change

(cherry picked from commit 78122e62614ca37f2a106da971e29e9c66fcbdc3)

Commit:
combine met emulator variants into a single plugin

(cherry picked from commit 6d654237d3a6967b3fb70ab626fb6d7be72950fb)

Commit:
make table sizes configurable

(cherry picked from commit 9d45304d9223572d0f178d7b97fbf323869a4997)

Commit:
add emulator with new types

(cherry picked from commit 95c3e2894faf421d98ba58eabed9595cdb9b5a02)

Commit:
cleanup comments

(cherry picked from commit f9e0c919fe3285ed799c506b41a95a77dc46ac7c)

Commit:
remove old version

(cherry picked from commit ab239bfa9824678e9a4a13975bf0dfe472d6ced5)

mtosi committed:
cleaning MTV

clean

Update MTVHistoProducerAlgoForTracker.cc

Update MTVHistoProducerAlgoForTracker.cc

fix code-format

fabiocos committed:
Rename MTDTopologyBuilder as MTDPixelTopologyBuilder

fabiocos committed:
Propagate ETL v5 start copy and offset into PMTDParameters

fabiocos committed:
Add ETL structure information to MTDTopology

fabiocos committed:
Redefine face layour using array, development commit

fabiocos committed:
horizontal and vertical shift functions

makortel committed:
Allow edmodule.clone(pset) for pset to override parameters

fabiocos committed:
Update topology, use for MTDDetSector navigation, ok for hshift

fabiocos committed:
Fixes to the vertical navigation

fabiocos committed:
Turn test of REcoMTD/DetLayers into a unit test, reference to be provided in Geometry/TestReference

fabiocos committed:
Remove useless argument in compatibleDetsLine

fabiocos committed:
Update for missing vshift, provide and use ifno about closest module in row if present

fabiocos committed:
Clean MEssageLogger in unit tests

fabiocos committed:
Silence debugging printout

felicepantaleo committed:
Use caloparticles when boundary crossed, otherwise use simclusters.

fabiocos committed:
Protect MTDTopology construction for legacy scenarios with 1 ETL disc

fabiocos committed:
scram b code-format

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33609/22447

ERROR: Build errors found during clang-tidy run.
```
L1Trigger/VertexFinder/plugins/VertexNTupler.cc:340:33: error: unknown type name 'TObjString' [clang-diagnostic-error]
      descriptionArray->Add(new TObjString(description.c_str()));
                                ^
Suppressed 1938 warnings (1933 in non-user code, 4 NOLINT, 1 with check filters).
--
gmake: *** [config/SCRAM/GMake/Makefile.coderules:128: code-checks] Error 2
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 2
```

Commit:
Merged l1t-AuantizedPuppiMet_120X from repository cecilecaillol with cms-merge-topic

Commit:
scram format

Commit:
code checks

cmsbuild committed:
Merge pull request #33614 from bsunanda/Run4-hgx285

Run4-hgx285 Update some of the remaining validation classes in Validation/HGCalValidation

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33609/22467

- This PR adds an extra 196KB to repository

- Found files with invalid states:
  - L1Trigger/VertexFinder/src/RecoVertexWithTP.cc:
    - Added: 6f253191085a9e0e20e15598648df1a7e41f9613
    - Modified: c2b74179b12744e49978375997717e735a23c271, 19fc3606939e98d33ed7addeb3cad8bff5741983
    - Deleted: cbe45b92f15c690a95a0acb04111b4194916fadf
  - L1Trigger/VertexFinder/interface/RecoVertexWithTP.h:
    - Added: 6f253191085a9e0e20e15598648df1a7e41f9613
    - Modified: c2b74179b12744e49978375997717e735a23c271
    - Deleted: cbe45b92f15c690a95a0acb04111b4194916fadf
  - L1Trigger/VertexFinder/src/L1TrackTruthMatched.cc:
    - Added: 6f253191085a9e0e20e15598648df1a7e41f9613
    - Modified: c2b74179b12744e49978375997717e735a23c271
    - Deleted: 30b5552f33f3ae4d50d7412e7dcd6ba6f3c74b7a

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DataFormats/L1Trigger/src/classes_def.xml  modified in PR(s): #33606

cmsbuild commented:
A new Pull Request was created by  @cecilecaillol  for master.

It involves the following packages:

DataFormats/L1Trigger
L1Trigger/VertexFinder
SimDataFormats/TrackingAnalysis
SimTracker/Common


The following packages do not have a category, yet:

L1Trigger/VertexFinder
Please create a PR for https://github.com/cms-sw/cms-bot/blob/master/categories_map.py to assign category

@cmsbuild, @rekovic, @civanch, @cecilecaillol, @mdhildreth can you please review it and eventually sign? Thanks.
@echabert, @pieterdavid, @robervalwalsh, @Martin-Grunewald, @thomreis, @threus, @mmusich, @cericeci, @kreczko, @makortel, @JanFSchulte, @jhgoh, @dgulhan, @apsallid, @prolay, @HuguesBrun, @trocino, @sscruz, @abbiendi, @GiacomoSguazzoni, @rovere, @VinInn, @ebrondol, @mtosi, @fabiocos, @gbenelli, @Fedespring, @calderona, @lecriste, @dinyar this is something you requested to watch as well.
@silviodonato, @dpiparo, @qliphy you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

cecilecaillol commented:
please test

smuzaffar commented on L1Trigger/VertexFinder/BuildFile.xml:
```diff
@@ -0,0 +1,23 @@
+<use name="SimTracker/TrackTriggerAssociation"/>
+<use name="boost"/>
+<use name="roothistmatrix"/>
+<flags CXXFLAGS="-g -Wno-unused-variable -Wno-misleading-indentation"/>
```
why these extra flags?

cecilecaillol replied:
Removed

smuzaffar commented on L1Trigger/VertexFinder/plugins/BuildFile.xml:
```diff
@@ -0,0 +1,29 @@
+<use name="CommonTools/UtilAlgos"/>
+<use name="CondFormats/DataRecord"/>
+<use name="L1Trigger/VertexFinder"/>
+<flags CXXFLAGS="-g -Wno-unused-variable"/>
```
please clean up these flags

cecilecaillol replied:
Done

Commit:
restore tau DQM for previous eras

Commit:
fix syntax of removal of idDecayModeOldDMs in DQM

Commit:
Merged l1t-alexx-VertexFinder_Jan_120X from repository cecilecaillol with cms-merge-topic

cmsbuild committed:
Merge pull request #33523 from pieterdavid/stripG2PCL_removeShallow

Merge ShallowGainCalibration logic in SiStripGainsPCLWorker and a minor bugfix

Commit:
clean flags

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33609/22470

- This PR adds an extra 196KB to repository

- Found files with invalid states:
  - L1Trigger/VertexFinder/src/RecoVertexWithTP.cc:
    - Added: 6f253191085a9e0e20e15598648df1a7e41f9613
    - Modified: c2b74179b12744e49978375997717e735a23c271, 19fc3606939e98d33ed7addeb3cad8bff5741983
    - Deleted: cbe45b92f15c690a95a0acb04111b4194916fadf
  - L1Trigger/VertexFinder/interface/RecoVertexWithTP.h:
    - Added: 6f253191085a9e0e20e15598648df1a7e41f9613
    - Modified: c2b74179b12744e49978375997717e735a23c271
    - Deleted: cbe45b92f15c690a95a0acb04111b4194916fadf
  - L1Trigger/VertexFinder/src/L1TrackTruthMatched.cc:
    - Added: 6f253191085a9e0e20e15598648df1a7e41f9613
    - Modified: c2b74179b12744e49978375997717e735a23c271
    - Deleted: 30b5552f33f3ae4d50d7412e7dcd6ba6f3c74b7a

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DataFormats/L1Trigger/src/classes_def.xml  modified in PR(s): #33606

cmsbuild commented:
Pull request #33609 was updated. @cmsbuild, @rekovic, @civanch, @cecilecaillol, @mdhildreth can you please check and sign again.

cecilecaillol commented:
please test

pieterdavid committed:
Use esConsumes for SiStripFedZeroSuppression

mtosi committed:
fix bug

cmsbuild committed:
Merge pull request #33595 from bsunanda/Run4-hgx282Z

Run4-hgx282Z Remove overlaps in the V15 geometry of HGCal

pieterdavid committed:
esConsumes for SiStripClusterToDigiProducer

pieterdavid committed:
esConsumes for SiStripHitEffFromCalibTree

cmsbuild commented:
-1

**Failed Tests**:   Build  ClangBuild 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b282c5/14846/summary.html
**COMMIT**: 0517c396f113354bb39f6cceded48bc6bdad4771
**CMSSW**: CMSSW_12_0_X_2021-05-03-2300/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/33609/14846/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Build

I found compilation warning when building:  See details on the summary page.

## Clang Build

I found compilation warning while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' scram build -k -j 32 COMPILER='llvm compile'

```
 See details on the summary page.

mtosi committed:
fix code-format

Commit:
import missing era modifiers to DQM config

cmsbuild committed:
Merge pull request #33474 from gartung/gartung-tbb2021

Changes required to compile with TBB 2021.2

Commit:
clean unused variable

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33609/22477

- This PR adds an extra 20KB to repository

- Found files with invalid states:
  - L1Trigger/VertexFinder/src/RecoVertexWithTP.cc:
    - Added: 6f253191085a9e0e20e15598648df1a7e41f9613
    - Modified: c2b74179b12744e49978375997717e735a23c271, 19fc3606939e98d33ed7addeb3cad8bff5741983
    - Deleted: cbe45b92f15c690a95a0acb04111b4194916fadf
  - L1Trigger/VertexFinder/interface/RecoVertexWithTP.h:
    - Added: 6f253191085a9e0e20e15598648df1a7e41f9613
    - Modified: c2b74179b12744e49978375997717e735a23c271
    - Deleted: cbe45b92f15c690a95a0acb04111b4194916fadf
  - L1Trigger/VertexFinder/src/L1TrackTruthMatched.cc:
    - Added: 6f253191085a9e0e20e15598648df1a7e41f9613
    - Modified: c2b74179b12744e49978375997717e735a23c271
    - Deleted: 30b5552f33f3ae4d50d7412e7dcd6ba6f3c74b7a

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DataFormats/L1Trigger/src/classes_def.xml  modified in PR(s): #33606

cmsbuild commented:
Pull request #33609 was updated. @cmsbuild, @rekovic, @civanch, @cecilecaillol, @mdhildreth can you please check and sign again.

cecilecaillol commented:
please test

Commit:
Cleanup of data/simulation analyzer modules

Commit:
Code format

Commit:
Move to test

justinasr committed:
Allow to suppress ExternalLHEProducer exception if not all events were processed

justinasr committed:
Merge branch 'master' into from-CMSSW_11_0_0_pre7-externallheproducer

mmusich committed:
fix missing leftover Timing parameter in fed_dqm_sourceclient-live_cfg.py

czangela committed:
remove remaining Timing parameters

czangela committed:
remove more Timing

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b282c5/14852/summary.html
**COMMIT**: 43beea08c27376ed58766494ee6c8f9e123a4f97
**CMSSW**: CMSSW_12_0_X_2021-05-04-1100/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/33609/14852/install.sh` to create a dev area with all the needed externals and cmssw changes.

**CMS Clang-Tidy warnings**: There are  Clang-Tidy warnings. See https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b282c5/14852/llvm-analysis/cmsclangtidy.txt for details.

## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 37
 - DQMHistoTests: Total histograms compared: 2662646
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2662623
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 36 files compared)
 - Checked 155 log files, 37 edm output root files, 37 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_0_X_2021-05-04-1100+b282c5/42614/triggerResults): no differences found

Commit:
Clean up a few classes for the use of ESGetToken and add a script for plots

Commit:
Code Check

justinasr committed:
Fix incorrect variable name

cmsbuild committed:
Merge pull request #33618 from perrotta/fixTypoInHGCRecHitNavigator

Fix typo in the fillDescriptions method of RecoParticleFlow/PFClusterProducer/interface/HGCRecHitNavigator.h

qliphy committed:
fix 33544

cecilecaillol commented:
@aperloff Does this rebase look fine to you?

cmsbuild committed:
Merge pull request #33625 from mmusich/fixMissingTiming

fix missing leftover Timing parameter in fed_dqm_sourceclient-live_cfg.py

Commit:
Initialize id variable to silence warning

justinasr committed:
Applied code checks patch

cmsbuild committed:
Merge pull request #33517 from jainshilpi/HtoZZULID

Adding HtoZZ UL ID configs

cmsbuild committed:
Merge pull request #33628 from qliphy/fixh7cocurrent

Set generateConcurrently to False to fix Herwig workflow issue

civanch committed:
updated physics configurations

cmsbuild committed:
Merge pull request #33488 from lecriste/MultiCluster_to_Trackster_120

[HGCAL] Deprecate MultiCluster

cmsbuild committed:
Merge pull request #33627 from bsunanda/Phase2-hgx285X

Phase2-hgx285X Clean up a few classes for the use of ESGetToken and add a script for plots

civanch committed:
code format

pieterdavid committed:
Fix compilation warning

cmsbuild committed:
Merge pull request #33573 from b-fontana/ecal_condobject_container_iterator_fix

EcalCondObject container iteration fix

cmsbuild committed:
Merge pull request #33582 from dildick/from-CMSSW_12_0_X_2021-04-19-2300-CSCValDQMReport

Cleanup of data/simulation analyzer modules in CSCTriggerPrimitives

cmsbuild committed:
Merge pull request #33631 from mrodozov/fix-warning-eventFilterUtils

Initialize id variable to silence warning

missirol committed:
avoiding trailing whitespaces in config dump

cmsbuild committed:
Merge pull request #33624 from justinasr/from-CMSSW_11_0_0_pre7-externallheproducer

Allow to suppress ExternalLHEProducer exception if not all events were processed

Commit:
Addition to PixelDQM: cluster charge corrected with template and also the template correction itself

Commit:
code-format patch

cecilecaillol commented:
+l1

makortel commented on DataFormats/L1Trigger/src/classes.h:
```diff
@@ -31,3 +31,11 @@
 #include "DataFormats/L1Trigger/interface/HOTPDigiTwinMux.h"
+#include "DataFormats/L1Trigger/interface/Vertex.h"
+
+namespace {
+  struct dictionary {
+    l1t::VertexCollection dummy0;
+    edm::Wrapper<l1t::VertexCollection> dummy1;
+  };
+}  // namespace
```
This is not needed since a while
```suggestion
```

makortel commented on L1Trigger/VertexFinder/interface/AnalysisSettings.h:
```diff
@@ -0,0 +1,95 @@
+  class AnalysisSettings : public AlgoSettings {
+  public:
+    AnalysisSettings(const edm::ParameterSet& iConfig);
+    ~AnalysisSettings() {}
```
Can there ever be a case that an `AnalysisSetting` object would be deleted via `AlgoSettings` pointer? I.e. effectively
```cpp
AlgoSettings *s = new AnalysisSettings();
delete s;
```

aperloff replied:
This use case was never envisioned by me, though I didn't write the original code. It's certainly not a use case I would see as useful, efficient, or recommended.

makortel replied:
Thanks, then this is ok. It's just that if there is a (even remote) possibility that an ownership to an object would be moved through a base class pointer, the destructor of the base class must be declared as `virtual`.

makortel commented on DataFormats/L1Trigger/interface/Vertex.h:
```diff
@@ -0,0 +1,34 @@
+
+  private:
+    float z0_;
+    std::vector<edm::Ptr<Track_t>> tracks_;
```
Just to double-check, can the `edm::Ptr<Track_t>` point to `Track_t` objects in different Event products? (if not, `edm::PtrVector<Track_t>` would be more efficient)

aperloff replied:
No, all of the `Track_t` objects should be from the same event.

makortel replied:
I meant two (or more) collections in the same event

aperloff replied:
It's possible. This doesn't happen right now, but I can come up with a contrived situation where you might be joining two `Track_t` collections and then forming a vertex (i.e. prompt and displaces). Unless we want to say that these sorts of contrived situations are not acceptable I would recommend leaving the code the way it is.

Commit:
Move DBTemplate access to token

makortel commented on L1Trigger/VertexFinder/interface/VertexFinder.h:
```diff
@@ -0,0 +1,126 @@
+    /// Number of reconstructed vertices
+    unsigned int numVertices() const { return vertices_.size(); }
+    /// Reconstructed Primary Vertex
+    RecoVertex<> PrimaryVertex() const {
```
Naming convention is to start method names with lower case letter.

makortel commented on L1Trigger/VertexFinder/interface/VertexProducer.h:
```diff
@@ -0,0 +1,49 @@
+  class AlgoSettings;
+}
+
+class VertexProducer : public edm::EDProducer {
```
Please use a thread-efficient base class
https://twiki.cern.ch/twiki/bin/view/CMSPublic/FWMultithreadedFrameworkModuleTypes

makortel replied:
Please also move the contents of this header to the corresponding source file (headers for modules are largely unnecessary).

makortel commented on L1Trigger/VertexFinder/plugins/VertexAnalyzer.cc:
```diff
@@ -0,0 +1,820 @@
+
+namespace l1tVertexFinder {
+
+  class VertexAnalyzer : public edm::EDAnalyzer {
```
Please use a thread-efficient base class.

makortel commented on L1Trigger/VertexFinder/plugins/VertexAnalyzer.cc:
```diff
@@ -0,0 +1,820 @@
+    // storage class for configuration parameters
+    AnalysisSettings settings_;
+
+    edm::Service<TFileService> fs_;
```
Service handles should not be stored as class member variables. Instead, instantiate it in each member function it is needed.

makortel commented on L1Trigger/VertexFinder/plugins/VertexAnalyzer.cc:
```diff
@@ -0,0 +1,820 @@
+      edm::ESHandle<TrackerGeometry> trackerGeometryHandle;
+      iSetup.get<TrackerDigiGeometryRecord>().get(trackerGeometryHandle);
+      edm::ESHandle<TrackerTopology> trackerTopologyHandle;
+      iSetup.get<TrackerTopologyRcd>().get(trackerTopologyHandle);
```
Please migrate to EventSetup consumes (as pointed to in the clang-tidy warnings https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b282c5/14852/llvm-analysis/cmsclangtidy.txt)

makortel commented on L1Trigger/VertexFinder/plugins/VertexAnalyzer.cc:
```diff
@@ -0,0 +1,820 @@
+    if (settings_.debug() > 2)
+      edm::LogInfo("VertexAnalyzer") << "analyzer::================ End of Event ==============";
+
+    delete RecoPrimaryVertex;
```
How about using a smart pointer?

makortel commented on L1Trigger/VertexFinder/plugins/VertexProducer.cc:
```diff
@@ -0,0 +1,123 @@
+
+  // Tame debug printout.
+  cout.setf(ios::fixed, ios::floatfield);
+  cout.precision(4);
```
Modules to be run in production should not alter `cout` like this.

makortel commented on L1Trigger/VertexFinder/plugins/VertexProducer.cc:
```diff
@@ -0,0 +1,123 @@
+
+  switch (settings_.vx_algo()) {
+    case Algorithm::FastHisto: {
+      edm::ESHandle<TrackerTopology> tTopoHandle = iSetup.getHandle(trackerTopologyToken_);
```
EventSetup consumes

makortel commented on L1Trigger/VertexFinder/src/InputData.cc:
```diff
@@ -0,0 +1,216 @@
+    }
+
+    // Get the tracker geometry info needed to unpack the stub info.
+    edm::ESHandle<TrackerGeometry> trackerGeometryHandle;
+    iSetup.get<TrackerDigiGeometryRecord>().get(trackerGeometryHandle);
+    const TrackerGeometry* trackerGeometry = trackerGeometryHandle.product();
+
+    edm::ESHandle<TrackerTopology> trackerTopologyHandle;
+    iSetup.get<TrackerTopologyRcd>().get(trackerTopologyHandle);
+    const TrackerTopology* trackerTopology = trackerTopologyHandle.product();
```
EventSetup consumes

makortel commented on SimDataFormats/TrackingAnalysis/interface/TrackingParticle.h:
```diff
@@ -173,12 +182,28 @@ class TrackingParticle {
   }
 
+  /// @brief dxy parameter. (This is the transverse impact parameter w.r.t. to (0,0,0) ONLY if refPoint is close to (0,0,0)).
+  double dxy() const { return (-vx() * py() + vy() * px()) / pt(); }
```
The comment is misleading because the `vx()` etc are the production point of the simulated particle. If the point is to get impact parameters with respect to the beamline or `(0,0,0)` I'd suggest to look [`ParametersDefinerForTP`](https://github.com/cms-sw/cmssw/blob/master/SimTracker/TrackAssociation/interface/ParametersDefinerForTP.h) that e.g. tracking validation uses for that purpose.

Commit:
code patch (again)

felicepantaleo committed:
modify Tracksters to have a float vertex multiplicity

felicepantaleo committed:
using float multiplicity for fraction

mseidel42 committed:
Allow for increased mass precision in NanoGen

Commit:
Merged l1t-AuantizedPuppiMet_120X from repository cecilecaillol with cms-merge-topic

aperloff commented:
> @aperloff Does this rebase look fine to you?

@cecilecaillol If, as you say, this is a rebase of https://github.com/cms-l1t-offline/cmssw/pull/874, then yes, it looks like the code at that point. However, remember that there have been some pretty significant revisions since then (https://github.com/cms-l1t-offline/cmssw/pull/901, https://github.com/cms-l1t-offline/cmssw/pull/902, https://github.com/cms-l1t-offline/cmssw/pull/909). Some of those revisions address the comments from @makortel (i.e. using a thread efficient EDProducer/EDAnalyzer and not using cout). I'm not sure what your plan is for bringing in those changes.

Commit:
address Silvio's comments

Commit:
code format

cecilecaillol commented:
> > @aperloff Does this rebase look fine to you?
> 
> @cecilecaillol If, as you say, this is a rebase of [cms-l1t-offline#874](https://github.com/cms-l1t-offline/cmssw/pull/874), then yes, it looks like the code at that point. However, remember that there have been some pretty significant revisions since then ([cms-l1t-offline#901](https://github.com/cms-l1t-offline/cmssw/pull/901), [cms-l1t-offline#902](https://github.com/cms-l1t-offline/cmssw/pull/902), [cms-l1t-offline#909](https://github.com/cms-l1t-offline/cmssw/pull/909)). Some of those revisions address the comments from @makortel (i.e. using a thread efficient EDProducer/EDAnalyzer and not using cout). I'm not sure what your plan is for bringing in those changes.

@aperloff I was rebasing the PRs progressively to avoid conflicts and large reviews. I will check if I manage to simultaneously rebase the PRs you mentioned to include the requested changes.

felicepantaleo committed:
add premix modifier

cmsbuild committed:
Merge pull request #33634 from missirol/devel_configTrailingWhitespaces

avoiding trailing whitespaces in config dump

hyunyong committed:
add default dPhi value

felicepantaleo committed:
add fraction cut

felicepantaleo committed:
set fraction cut to 0.05

Commit:
Fix the SiPixelTemplateDBObject Token with the beginRun transition

Commit:
Fix the SiPixelTemplateDBObject Token with the beginRun transition

fabiocos committed:
Adapt gap parameters for old ETL scenarios, where it was not used (D49 is based on I10, thanks to Pablo)

parbol committed:
MTDparameters.xml updated for compatibility with old geometries

Commit:
code-format patch

cecilecaillol commented:
@aperloff I cannot merge the following PRs easily (conflicts, may need other L1 PRs to be ported first). Can we merge this part first? Or can you take care of porting some of the necessary changes?

jeongeun committed:
change default cfi name produceMuons to muonSeedGenerator

civanch committed:
clean-up user actions

felicepantaleo committed:
disable fraction cut

davidlange6 committed:
move unit tests to python3

parbol committed:
Update mtdParameters for D49 and D60 (Bug fix)

mmusich committed:
TrackDistanceValueMapProducer: check the size of the reference vector before copying

Commit:
Reduce the number of plots generated with tempalte info, move the TrackerGeometry and Topology loading at BeginRun

sroychow committed:
remove Fall17V1 variables from electron table

sroychow committed:
update dqm according to fall17v1 removal

fabiocos committed:
Update mtd validation configuration to scenario D76, use up-to-date fragments and options

aperloff commented:
> @aperloff I cannot merge the following PRs easily (conflicts, may need other L1 PRs to be ported first). Can we merge this part first? Or can you take care of porting some of the necessary changes?

@cecilecaillol I'm okay if you go ahead with this PR and do the others later. I just wanted to make you aware that some of the requested changes were already implemented in the cms-l1t-offline fork.

Commit:
Some code semplification applied

Commit:
Merged l1t-alexx-VertexFinder_Jan_120X from repository cecilecaillol with cms-merge-topic

Commit:
Merged l1t-Emily_TkObjects_120X from repository cecilecaillol with cms-merge-topic

Commit:
code format

Commit:
move nuplizer files to correct directory

Commit:
remove unused variable

Commit:
Merged l1t-Emily_TkObjects_120X from repository cecilecaillol with cms-merge-topic

lecriste committed:
Avoid duplicate plots with 'all' option

Commit:
reorganize files

swagata87 committed:
move EgammaLocalCovParamDefaults.h from RecoEgamma to RecoEcal package

cmsbuild committed:
Merge pull request #33528 from cms-tau-pog/CMSSW_11_3_X_tau-pog_boostedTaus

Add boosted taus to NanoAOD

cmsbuild committed:
Merge pull request #33513 from cms-tau-pog/CMSSW_11_3_X_tau-pog_nanoAODcleanUp

Clean up tau related nanoAOD content

cmsbuild committed:
Merge pull request #33603 from ggovi/condcore-lumicondforhlt-7-120X

Bug fix for tag lock clean up at session closing time

cmsbuild committed:
Merge pull request #33649 from fabiocos/fc-mtdvalidation

MTD validation: update standalone test configuration to scenario D76, use up-to-date fragments

cmsbuild committed:
Merge pull request #33647 from parbol/MTDParametersUpdate

Mtd parameters update for D49 and D60 geometries (Bug fix)

Commit:
typo fix

lecriste committed:
Move plugins out of SimDataFormats

Commit:
lower case for methods, smart pointer

lecriste committed:
Uniform LCToCP naming

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33609/22542

- This PR adds an extra 36KB to repository

- Found files with invalid states:
  - L1Trigger/VertexFinder/src/RecoVertexWithTP.cc:
    - Added: 6f253191085a9e0e20e15598648df1a7e41f9613
    - Modified: c2b74179b12744e49978375997717e735a23c271, 19fc3606939e98d33ed7addeb3cad8bff5741983
    - Deleted: cbe45b92f15c690a95a0acb04111b4194916fadf
  - L1Trigger/VertexFinder/interface/RecoVertexWithTP.h:
    - Added: 6f253191085a9e0e20e15598648df1a7e41f9613
    - Modified: c2b74179b12744e49978375997717e735a23c271
    - Deleted: cbe45b92f15c690a95a0acb04111b4194916fadf
  - L1Trigger/VertexFinder/src/L1TrackTruthMatched.cc:
    - Added: 6f253191085a9e0e20e15598648df1a7e41f9613
    - Modified: c2b74179b12744e49978375997717e735a23c271
    - Deleted: 30b5552f33f3ae4d50d7412e7dcd6ba6f3c74b7a

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DataFormats/L1Trigger/src/classes_def.xml  modified in PR(s): #33606

Code check has found code style and quality issues which could be resolved by applying following patch(s)

- **code-format**:
https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33609/22542/code-format.patch
e.g. `curl https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33609/22542/code-format.patch | patch -p1`
You can also run `scram build code-format` to apply code format directly

Commit:
thread safe producer

tlampen committed:
Beamspot and vertex smaring for 2017 pp ref (forward port of #33504)

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33609/22543

- This PR adds an extra 36KB to repository

- Found files with invalid states:
  - L1Trigger/VertexFinder/src/RecoVertexWithTP.cc:
    - Added: 6f253191085a9e0e20e15598648df1a7e41f9613
    - Modified: c2b74179b12744e49978375997717e735a23c271, 19fc3606939e98d33ed7addeb3cad8bff5741983
    - Deleted: cbe45b92f15c690a95a0acb04111b4194916fadf
  - L1Trigger/VertexFinder/interface/RecoVertexWithTP.h:
    - Added: 6f253191085a9e0e20e15598648df1a7e41f9613
    - Modified: c2b74179b12744e49978375997717e735a23c271
    - Deleted: cbe45b92f15c690a95a0acb04111b4194916fadf
  - L1Trigger/VertexFinder/src/L1TrackTruthMatched.cc:
    - Added: 6f253191085a9e0e20e15598648df1a7e41f9613
    - Modified: c2b74179b12744e49978375997717e735a23c271
    - Deleted: 30b5552f33f3ae4d50d7412e7dcd6ba6f3c74b7a

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DataFormats/L1Trigger/src/classes_def.xml  modified in PR(s): #33606

Code check has found code style and quality issues which could be resolved by applying following patch(s)

- **code-format**:
https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33609/22543/code-format.patch
e.g. `curl https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33609/22543/code-format.patch | patch -p1`
You can also run `scram build code-format` to apply code format directly

fabiocos committed:
updates addressing comments by C. Vuosalo

fabiocos committed:
scram b code-format

Commit:
remove ESHandle

Commit:
code format

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33609/22547

- This PR adds an extra 48KB to repository

- Found files with invalid states:
  - L1Trigger/VertexFinder/src/RecoVertexWithTP.cc:
    - Added: 6f253191085a9e0e20e15598648df1a7e41f9613
    - Modified: c2b74179b12744e49978375997717e735a23c271, 19fc3606939e98d33ed7addeb3cad8bff5741983
    - Deleted: cbe45b92f15c690a95a0acb04111b4194916fadf
  - L1Trigger/VertexFinder/interface/RecoVertexWithTP.h:
    - Added: 6f253191085a9e0e20e15598648df1a7e41f9613
    - Modified: c2b74179b12744e49978375997717e735a23c271
    - Deleted: cbe45b92f15c690a95a0acb04111b4194916fadf
  - L1Trigger/VertexFinder/src/L1TrackTruthMatched.cc:
    - Added: 6f253191085a9e0e20e15598648df1a7e41f9613
    - Modified: c2b74179b12744e49978375997717e735a23c271
    - Deleted: 30b5552f33f3ae4d50d7412e7dcd6ba6f3c74b7a

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DataFormats/L1Trigger/src/classes_def.xml  modified in PR(s): #33606

Code check has found code style and quality issues which could be resolved by applying following patch(s)

- **code-format**:
https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33609/22547/code-format.patch
e.g. `curl https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33609/22547/code-format.patch | patch -p1`
You can also run `scram build code-format` to apply code format directly

Commit:
code format

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33609/22548

- This PR adds an extra 48KB to repository

- Found files with invalid states:
  - L1Trigger/VertexFinder/src/RecoVertexWithTP.cc:
    - Added: 6f253191085a9e0e20e15598648df1a7e41f9613
    - Modified: c2b74179b12744e49978375997717e735a23c271, 19fc3606939e98d33ed7addeb3cad8bff5741983
    - Deleted: cbe45b92f15c690a95a0acb04111b4194916fadf
  - L1Trigger/VertexFinder/interface/RecoVertexWithTP.h:
    - Added: 6f253191085a9e0e20e15598648df1a7e41f9613
    - Modified: c2b74179b12744e49978375997717e735a23c271
    - Deleted: cbe45b92f15c690a95a0acb04111b4194916fadf
  - L1Trigger/VertexFinder/src/L1TrackTruthMatched.cc:
    - Added: 6f253191085a9e0e20e15598648df1a7e41f9613
    - Modified: c2b74179b12744e49978375997717e735a23c271
    - Deleted: 30b5552f33f3ae4d50d7412e7dcd6ba6f3c74b7a

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DataFormats/L1Trigger/src/classes_def.xml  modified in PR(s): #33606

cmsbuild commented:
Pull request #33609 was updated. @cmsbuild, @rekovic, @civanch, @cecilecaillol, @mdhildreth can you please check and sign again.

cecilecaillol commented:
please test

Commit:
Added specific plot for Inner/outer modules, unuseful plot removed, disabled these plots on Online

cmsbuild commented:
-1

**Failed Tests**:   Build  HeaderConsistency  ClangBuild 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b282c5/14938/summary.html
**COMMIT**: 8835d71c407faf975b4a691aff585e021fb3e9c5
**CMSSW**: CMSSW_12_0_X_2021-05-06-2300/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/33609/14938/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Build

I found compilation warning when building:  See details on the summary page.

## Clang Build

I found compilation warning while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' scram build -k -j 64 COMPILER='llvm compile'

```
 See details on the summary page.

cmsbuild committed:
Merge pull request #33483 from hqucms/dev/particle_net_mass_regression

Add ParticleNet mass regression

cmsbuild committed:
Merge pull request #33639 from lecriste/makeHGCalValidationPlots_simplify_master

[HGCAL] Simplify makeHGCalValidationPlots

cmsbuild committed:
Merge pull request #33617 from dildick/from-CMSSW_12_0_X_2021-05-03-1100-GEMCSC-Run3-prep-2

More prep work for the Run-3 GEM-CSC integrated local trigger

cmsbuild committed:
Merge pull request #33599 from theofil/renamingCFGs

Renaming cfgs and cffs

gennai committed:
added the possibility to lock a record

ggovi committed:
Added support of Tag access restrictions

casarsa committed:
Added optional histograms of BTL and ETL UncalibratedRecHits to monitor the BTL and ETL time resolutions and a ROOT macro to analyze them.

Commit:
remove unused variables

cecilecaillol commented:
please test

gennai committed:
fixed the compilation error

casarsa committed:
Applied code-format and renamed macro to macros

casarsa committed:
Removed the macro folder

casarsa committed:
Applied code-format

pieterdavid committed:
Avoid 'handle' in member names that are not ESHandles anymore

lecriste committed:
Uniform LCToSC naming

lecriste committed:
Uniform TSToSC naming

lecriste committed:
Forgot auto

lecriste committed:
code-format

lecriste committed:
format code-format

cmsbuild commented:
-1

**Failed Tests**:   HeaderConsistency 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b282c5/14945/summary.html
**COMMIT**: df6ab6ae91a0f28f137b2a05a386f32597d9f687
**CMSSW**: CMSSW_12_0_X_2021-05-06-2300/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/33609/14945/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 4 differences found in the comparisons
 - DQMHistoTests: Total files compared: 37
 - DQMHistoTests: Total histograms compared: 2662646
 - DQMHistoTests: Total failures: 7
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2662617
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 36 files compared)
 - Checked 155 log files, 37 edm output root files, 37 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_0_X_2021-05-06-2300+b282c5/42685/triggerResults): no differences found

cmsbuild committed:
Merge pull request #33635 from arossi83/PixTemplateCorr

Addition to PixelDQM of Template correction to cluster charge

cmsbuild committed:
Merge pull request #33633 from civanch/geant4_physics_for_10.7

Updated physics configurations for Geant4

fabiocos committed:
move static constexpr as class member in MTDTopology

mmusich committed:
post PR #33635 merge fixes

Commit:
fix header

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33609/22569

- This PR adds an extra 228KB to repository

- Found files with invalid states:
  - L1Trigger/VertexFinder/src/RecoVertexWithTP.cc:
    - Added: 6f253191085a9e0e20e15598648df1a7e41f9613
    - Modified: c2b74179b12744e49978375997717e735a23c271, 19fc3606939e98d33ed7addeb3cad8bff5741983
    - Deleted: cbe45b92f15c690a95a0acb04111b4194916fadf
  - L1Trigger/VertexFinder/interface/RecoVertexWithTP.h:
    - Added: 6f253191085a9e0e20e15598648df1a7e41f9613
    - Modified: c2b74179b12744e49978375997717e735a23c271
    - Deleted: cbe45b92f15c690a95a0acb04111b4194916fadf
  - L1Trigger/VertexFinder/src/L1TrackTruthMatched.cc:
    - Added: 6f253191085a9e0e20e15598648df1a7e41f9613
    - Modified: c2b74179b12744e49978375997717e735a23c271
    - Deleted: 30b5552f33f3ae4d50d7412e7dcd6ba6f3c74b7a

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DataFormats/L1Trigger/src/classes_def.xml  modified in PR(s): #33606

cmsbuild commented:
Pull request #33609 was updated. @cmsbuild, @rekovic, @civanch, @cecilecaillol, @mdhildreth can you please check and sign again.

cecilecaillol commented:
please test

mseidel42 committed:
Increase mass precision for SUSY particles

cmsbuild committed:
Merge pull request #33669 from mmusich/postMergeFix

Pixel DQM: post PR #33635 merge fixes

jeongeun committed:
replace EDProducer with clone from the default cipython and cleanup

cmsbuild committed:
Merge pull request #33638 from felicepantaleo/SimTrackstersFromCP_12_0_X

[HGCAL] SimTracksters from Caloparticles

cmsbuild committed:
Merge pull request #33662 from casarsa/mc-mtdValidationUpdate

MTD validation: optional histograms of BTL and ETL uncalibrated RecHits

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b282c5/14957/summary.html
**COMMIT**: 49f0bfc1f320711d42bb4b2810f074f0ad0d3702
**CMSSW**: CMSSW_12_0_X_2021-05-09-2300/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/33609/14957/install.sh` to create a dev area with all the needed externals and cmssw changes.

Found compilation warnings

## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 6 differences found in the comparisons
 - DQMHistoTests: Total files compared: 37
 - DQMHistoTests: Total histograms compared: 2663174
 - DQMHistoTests: Total failures: 13
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2663139
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 36 files compared)
 - Checked 155 log files, 37 edm output root files, 37 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_0_X_2021-05-09-2300+b282c5/42694/triggerResults): no differences found

ggovi committed:
Fix for incomplete query!

mmusich committed:
move from orbitNumber() to luminosityBlock() in the simulation case

cmsbuild committed:
Merge pull request #33551 from hyunyong/gemStripTopoPR_CMSSW_12_0_X_2021-04-27-1100

Remove hard-coded dphi value

Commit:
Subdirectories for GEMCSC LUTs

justinasr committed:
Use std::getenv in ExternalLHEProducer

Commit:
Add possibility of studying MB in all subvolumes of CMS

Commit:
Consolidate readmes

Commit:
Document slope value

Commit:
Code check

Commit:
Turn-off GEM-CSC integrated local trigger

cmsbuild committed:
Merge pull request #33646 from davidlange6/dl210506

move another set of python tool unit tests to python3

cmsbuild committed:
Merge pull request #33600 from bsunanda/Run3-gex73

Run3-gex73 Remove reference of DDD from plugins/dd4hep/DDEcalEndcapAlgo.cc

lecriste committed:
Static Check

lecriste committed:
code-reformat

cmsbuild committed:
Merge pull request #33678 from dildick/from-CMSSW_12_0_X_2021-05-10-1100-disable-GEM-CSC-integrated-local-trigger

Turn-off GEM-CSC integrated local trigger in valCscStage2Digis [12_0_X]

mmusich committed:
update documentation after commit 69c8898814ce3be38134d8a3a5bcea09951a537c

Commit:
Improve CSC TP quality control

Commit:
Use .length instead of strlen() in AMPTHadronizer

Resolves: https://github.com/cms-sw/cmssw/issues/33682
(or at least will try to)

Commit:
Update AMPTHadronizer.cc

cmsbuild committed:
Merge pull request #33674 from justinasr/from-CMSSW_12_0_X_2021-05-10-1100

Use std::getenv in ExternalLHEProducer

cmsbuild committed:
Merge pull request #33685 from mmusich/updateDocsInTrackerGeometryBuilder

 Geometry/TrackerGeometryBuilder: update documentation after #27448

cmsbuild committed:
Merge pull request #33637 from mseidel42/NanoGen_120

Allow for increased mass precision in NanoGen + increase mass precision of SUSY particles in NanoAod

cmsbuild committed:
Merge pull request #33632 from pieterdavid/sistrip_esconsumes_3

esConsumes migration for SiStripFedZeroSuppression, SiStripHitEffFromCalibTree, and SiStripClusterToDigiProducer

cmsbuild committed:
Merge pull request #33673 from makortel/clonePSet

Allow edmodule.clone(pset) for PSet to override parameters

cmsbuild committed:
Merge pull request #33655 from tlampen/beamspot-and-vtxsmearing-for-ppRefULMC

Beamspot and vertex smaring for 2017 pp ref (forward port of #33504)

cmsbuild committed:
Merge pull request #33621 from PFCal-dev/hgc-tpg-integration-210504

[HGCAL trigger] Fixing phi smoothing in cluster seeding

civanch commented on SimTracker/Common/interface/TrackingParticleSelector.h:
```diff
@@ -64,64 +64,94 @@ class TrackingParticleSelector {
+      }
+    }
+    // test for remaining unstabled due to lack of genparticle pointer
+    auto pdgid = tp->pdgId();
```
use int pdgid = std::abs(tp->pdgId()); and retmove std::abs below

cecilecaillol replied:
Fixed

cmsbuild committed:
Merge pull request #33664 from lecriste/movePlugins

[HGCAL] Move plugins out of SimDataFormats

cmsbuild committed:
Merge pull request #33675 from bsunanda/Run3-gex74

Run3-gex74 Add possibility of studying MB in all subvolumes of CMS

mmusich committed:
migrate SLHCUpgradeSimulations/Geometry to esConsumes

dmeuser committed:
Added 2D maps of Impact Parameters for TkDQM

cmsbuild committed:
Merge pull request #33648 from mmusich/fixTrackDistanceValueMapProducer

TrackDistanceValueMapProducer: check the size of the reference vector before copying

cmsbuild committed:
Merge pull request #33672 from mmusich/SiStripMonitorClustermoveFromOrbitToLS

SiStripMonitorCluster: move from orbitNumber() to luminosityBlock()

cmsbuild committed:
Merge pull request #33606 from cecilecaillol/l1t-Emyr_120X

L1T phase-2: Update to histogram (Phase-1 like) jets

dmeuser committed:
Applied code format

cmsbuild committed:
Merge pull request #33428 from fwyzard/auto_gpu_workflows

Redesign all GPU workflows to detect if a GPU is present, and fall back to CPU otherwise

cmsbuild committed:
Merge pull request #33687 from mrodozov/patch-9

Use .length instead of strlen() in AMPTHadronizer

cmsbuild committed:
Merge pull request #33686 from dildick/from-CMSSW_12_0_X_2021-05-10-1100-more-lct-quality-control

Improve quality control for CSC trigger primitives

cmsbuild committed:
Merge pull request #33500 from mtosi/MTVcleaning

MTV fast cleaning

Dr15Jones committed:
Split ProductProvenanceRetriever into to classes

ProductProvenanceLookup is the API intended for accessing the information.
ProductProvenanceRetriever now just handles the retrieving of data from storage.

mmusich committed:
PrimaryVertexMonitoring: don't use HP flag for Alignment plots at HLT

cmsbuild committed:
Merge pull request #33626 from czangela/fix_sipixelrawtodigi_04_05_2021

Fix leftover Timing parameters

cmsbuild committed:
Merge pull request #33653 from swagata87/fixCircularDep_12XY

Fix circular dependency between RecoEgamma and RecoEcal

cmsbuild committed:
Merge pull request #33563 from jeongeun/updatemuon113X

Migrate module config in RecoMuon to use automatically generated cfipython default

Dr15Jones committed:
Moved ProductProvenanceRetriever to FWCore/Framework

This allows DataFormats/Provenance to no longer depend upon FWCore/Concurrency.

sharmaaash committed:
Update QQToHToTauTau_mh125_TuneCP5_14TeV_pythia8_cfi.py

sharmaaash committed:
Update TTbarToDilepton_mt172p5_TuneCP5_14TeV_pythia8_cfi.py

sroychow committed:
remove fall17v1 variable from nanoDQM

ianna committed:
as suggested by Vincenzo

Commit:
address comments from civanch

cecilecaillol commented on SimDataFormats/TrackingAnalysis/interface/TrackingParticle.h:
```diff
@@ -173,12 +182,28 @@ class TrackingParticle {
+
+  /// @brief dz parameter (= dsz/cos(lambda)). This is the track z0 w.r.t (0,0,0) only if the refPoint is close to (0,0,0).
+  double dz() const {
+    const auto thept2inv = 1 / p4().Perp2();
```
Fixed

jeongeun committed:
remove deprecated parameters

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33609/22622

- This PR adds an extra 224KB to repository

- Found files with invalid states:
  - L1Trigger/VertexFinder/src/RecoVertexWithTP.cc:
    - Added: 6f253191085a9e0e20e15598648df1a7e41f9613
    - Modified: c2b74179b12744e49978375997717e735a23c271, 19fc3606939e98d33ed7addeb3cad8bff5741983
    - Deleted: cbe45b92f15c690a95a0acb04111b4194916fadf
  - L1Trigger/VertexFinder/interface/RecoVertexWithTP.h:
    - Added: 6f253191085a9e0e20e15598648df1a7e41f9613
    - Modified: c2b74179b12744e49978375997717e735a23c271
    - Deleted: cbe45b92f15c690a95a0acb04111b4194916fadf
  - L1Trigger/VertexFinder/src/L1TrackTruthMatched.cc:
    - Added: 6f253191085a9e0e20e15598648df1a7e41f9613
    - Modified: c2b74179b12744e49978375997717e735a23c271
    - Deleted: 30b5552f33f3ae4d50d7412e7dcd6ba6f3c74b7a

cmsbuild commented:
Pull request #33609 was updated. @cmsbuild, @rekovic, @civanch, @cecilecaillol, @mdhildreth can you please check and sign again.

cecilecaillol commented:
please test

Commit:
Remove tests for root_numpy and root_pandas

mmusich committed:
introduce DiMuonVertexValidation in Alignment validation suite

Commit:
remove file testRootNumpy.py

cavana committed:
implements UPT invariant mass triggers in uGT emulator

cmsbuild committed:
Merge pull request #33696 from mmusich/noHPforHLTTracksPVAlignmentMonitoring

PrimaryVertexMonitoring: don't use HP flag for Alignment plots at HLT

jeongeun committed:
remove deprecated parameters

Commit:
fixes by Emily

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b282c5/15024/summary.html
**COMMIT**: 46390b69d6d63ba8d8f9faee75384486b15469ce
**CMSSW**: CMSSW_12_0_X_2021-05-11-2300/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/33609/15024/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 4 differences found in the comparisons
 - DQMHistoTests: Total files compared: 37
 - DQMHistoTests: Total histograms compared: 2648114
 - DQMHistoTests: Total failures: 6
 - DQMHistoTests: Total nulls: 1
 - DQMHistoTests: Total successes: 2648085
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: -0.004 KiB( 36 files compared)
 - DQMHistoSizes: changed ( 312.0 ): -0.004 KiB       MessageLogger/Warnings
 - Checked 155 log files, 37 edm output root files, 37 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_0_X_2021-05-11-2300+b282c5/42747/triggerResults): no differences found

cecilecaillol commented:
+l1

sroychow committed:
add pre ul era modifiers correctly

cmsbuild committed:
Merge pull request #33693 from dmeuser/TkDQM_2Dmaps_ImpactParamter

2D Maps of Impact Parameters for TkDQM

cavana committed:
resolves style issues

Commit:
Fix compilation failure in DBG in RecoPixelVertexing/PixelVertexFinding

Martin-Grunewald committed:
HLT menu migration to 12_0_0_pre1

Commit:
remove useless cfi

Commit:
minor comment from Silvio

perrotta committed:
Deprecated parameter and header cleaning

sroychow committed:
fix es data access

ianna committed:
update unit test

cmsbuild committed:
Merge pull request #33651 from cecilecaillol/l1t-Emily_TkObjects_120X

L1T Phase-2: add various L1 Tk objects

cmsbuild committed:
Merge pull request #33677 from dildick/from-CMSSW_12_0_X_2021-05-10-1100-GEMCSC-LUTs

Subdirectories for GEM-CSC LUTs

cmsbuild committed:
Merge pull request #33659 from MilanoBicocca-pix/12_0_X_BeamSpot_LockRecords

 added the possibility to lock a record

cmsbuild committed:
Merge pull request #33619 from cecilecaillol/l1t-AuantizedPuppiMet_120X

L1T Phase-2: Quantized Puppi MET emulators

cmsbuild committed:
Merge pull request #33671 from jeongeun/recoecal113X

Migrate module config in RecoEcal to use default cfipython

cecilecaillol committed:
L1T: add HPS taus (#33681)

* Uploaded HPS Tau algorithm for new PR

(cherry picked from commit 8f55d733315824bd1c5d04c2888bbd322cc6a2ac)

* modified considering Andrea's comments

(cherry picked from commit a30f47c363191b2d39c0ff137b286d84d9770478)

* DataFormats/Phase2L1Taus is merged with DataFormats/L1TParticleFlow

(cherry picked from commit 21829a0a5bc3c2f84025848d37d1f570ef38625a)

* renamed HPSPFTauBuilder and HPSPFTauQualityCut as original

(cherry picked from commit 391723c56fcdb1cb18bbc8e53a07cd4346f1d576)

* code format

* code format

* fix BuildFile

* remove cout and switch to CaloJet

Co-authored-by: sandeepbhowmik1 <sandeep.bhowmik@cern.ch>
Co-authored-by: ccaillol <ccaillol@cern.ch>

cmsbuild committed:
Merge pull request #33720 from cms-tsg-storm/HLTmigrationTo1200pre1

HLT Menu Migration to 12_0_0_pre1

cmsbuild committed:
Merge pull request #33689 from mmusich/esConsumesSLHCUpgradeSimulations

migrate SLHCUpgradeSimulations/Geometry to esConsumes

Dr15Jones committed:
Removed unnecessary includes

cmsbuild committed:
Merge pull request #33722 from perrotta/removeDepecatedTimingParameter

Remove a deprecated "Timing" parameter from SiPixelDigiToRaw (and some header cleaning)

cmsbuild committed:
Merge pull request #33701 from sroychow/electronsremovefall17V1

EGamma: Remove Fall17V1 variables from electron table for v9 NanoAOD

Commit:
change the unsigned to int and keep the int as it was

silviodonato commented:
@cms-sw/simulation-l2 do you have futher comments?

silviodonato commented on DataFormats/L1Trigger/src/classes_def.xml:
```diff
@@ -278,4 +278,10 @@
+
+  <class name="l1t::Vertex"/>
+  <class name="std::vector<l1t::Vertex>"/>
+  <class name="l1t::VertexCollection"/>
```
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b282c5/15024/dupDict/lostDefs.txt

```
Searching for 'classes_def.xml' in '/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-05-11-2300/src'.

./DataFormats/L1Trigger/src/classes_def.xml
  Vertex : l1t::Vertex
  Vertex : vector< l1t::Vertex >
```

Looking at the example in https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideCreatingNewProducts#Create_a_classes_def_xml_in_the
I think you have to remove `  <class name="l1t::VertexCollection"/>` as it is already defined in `  <class name="std::vector<l1t::Vertex>"/>`

Please note that in the other l1t data formats there is the `BxCollection` version of the data format. In this case it would be `l1t::VertexBxCollection`, but I see that it is not implemented in the code.

cmsbuild committed:
Merge pull request #33707 from mrodozov/remove-rootnumpy-rootpandas

Remove tests for root_numpy and root_pandas

cmsbuild committed:
Merge pull request #33708 from cavana/upt-invariant-mass_12_0_X

implements UPT invariant mass triggers in uGT emulator

cmsbuild committed:
Merge pull request #33700 from sharmaaash/master

Updated import TUNECP5

ggovi committed:
Added support of "frequent" flag/ stored timestamps moved to UTC

fabiocos committed:
Remove forgotten old commented statements (comment by P. Srimanobhas)

cmsbuild committed:
Merge pull request #33721 from sroychow/fixesaccessnano

NanoAOD:migrate modules to ESConsumes

cmsbuild committed:
Merge pull request #33660 from ggovi/condcore-conddb-tools-tag-authorization-1-120X

Setting of Tag access restrictions in command line tools

cmsbuild committed:
Merge pull request #33709 from mmusich/DiMuonVertexValidation_in_CMSSW

introduce DiMuonVertexValidation in Alignment validation suite

cmsbuild committed:
Merge pull request #33699 from Dr15Jones/ProductProvenanceRetriever

Moved ProductProvenanceRetriever to FWCore/Framework

cmsbuild committed:
Merge pull request #33705 from ianna/math-units-return-double

user-defined floating-point literals return double instead of long double

cmsbuild committed:
Merge pull request #33733 from ggovi/condcore-utilities-o2o-8-120X

Improvements of O2O steering tools

cmsbuild committed:
Merge pull request #33718 from mrodozov/RecoPixelVertexing-PixelVertexFinding-comp-failure

Fix compilation failure in DBG in RecoPixelVertexing/PixelVertexFinding

cmsbuild committed:
Merge pull request #33598 from fabiocos/fc-etllayout

MTD geometry and reconstruction: store ETL layout logic into MTDTopology, use for ETL layers navigation in reconstruction

Commit:
Merged l1t-alexx-VertexFinder_Jan_120X from repository cecilecaillol with cms-merge-topic

Commit:
applied silvio's comment

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33609/22700

- This PR adds an extra 16KB to repository

- Found files with invalid states:
  - L1Trigger/TrackFindingTracklet/test/L1TrackObjectNtupleMaker.cc:
    - Added: ba6c6b9e11ec3e8cbd5aae66daaf0a6ec9dd6f62
    - Modified: 987f7956ae6adde84a9f94d648ec4df86888bfaa
    - Deleted: 5d86491a71ca1dace4fe86de80c55ee9c0b00448
  - Validation/MtdValidation/macro/makeTimeResPlots.C:
    - Added: defee1c12a701204ddb88b11664260f8538493d2
    - Deleted: d3b4d1251b8e4ffcdcf56c4c9b4c3d0fd1fbb676
  - L1Trigger/TrackFindingTracklet/test/L1TrackObjectNtupleMaker_cfg.py:
    - Added: ba6c6b9e11ec3e8cbd5aae66daaf0a6ec9dd6f62
    - Deleted: 5d86491a71ca1dace4fe86de80c55ee9c0b00448
  - L1Trigger/Phase2L1ParticleFlow/plugins/L1MetPfProducerNewTypes.cc:
    - Added: b83fdbb07569a14b5baee721dffdf3e89da5b56c
    - Deleted: 9dcba0ef87ceff4603fa9308a4d10ad14ff12165
  - L1Trigger/L1TTrackMatch/plugins/L1TrackJetProducer.h:
    - Added: ba6c6b9e11ec3e8cbd5aae66daaf0a6ec9dd6f62
    - Modified: 976a1aec7b3bc41f94366928a521703a48141238, 987f7956ae6adde84a9f94d648ec4df86888bfaa
    - Deleted: 4e61998b2188fda1b744d7a17af2bef6e82d2cd0
  - L1Trigger/Phase2L1ParticleFlow/plugins/L1MetPfProducerTDR.cc:
    - Added: 9dcba0ef87ceff4603fa9308a4d10ad14ff12165
    - Deleted: bc6ed5f7f0a8c733e5285d43867e08b12f77ceda
  - L1Trigger/Phase2L1ParticleFlow/python/L1METPFProducer_cfi.py:
    - Added: 5c055aad6dbd7b68e9f96afb165f0ab93768bf9a
    - Deleted: a3612b9f645f090936f5e19db877781d8ec0ec1b

cmsbuild commented:
Pull request #33609 was updated. @SiewYan, @andrius-k, @gouskos, @chayanit, @wajidalikhan, @ianna, @kpedro88, @ErnestaP, @Martin-Grunewald, @rekovic, @tlampen, @alberto-sanchez, @pohsun, @santocch, @cecilecaillol, @perrotta, @civanch, @yuanchao, @makortel, @srimanob, @ahmad3213, @cmsbuild, @agrohsje, @fwyzard, @GurpreetSinghChahal, @smorovic, @davidlange6, @smuzaffar, @Dr15Jones, @jpata, @cvuosalo, @mdhildreth, @jfernan2, @slava77, @ggovi, @qliphy, @fabiocos, @francescobrivio, @malbouis, @mkirsano, @jordan-martins, @kmaeshima, @emeschi, @silviodonato, @franzoni, @fgolf, @mariadalfonso, @rvenditti can you please check and sign again.
